lib/job/arBaseJob.class.php

```diff
@@ -44,12 +44,13 @@ class arBaseJob extends Net_Gearman_Job_Common
    * arObjectMoveJob and the import jobs. The Gearman job server doesn't include a built in
    * system to postpone/schedule jobs so, if multiple jobs from the $avoidParallelExecutionJobs
    * variable bellow are executed at the same time, the late ones will wait, retrying after
-   * the amount of seconds indicated in $waitForRetryTime, until the previous ones are finished.
-   * Due to the limitations of the Gearman job server, the waiting jobs will block the workers
-   * executing them until they are finished.
+   * the amount of seconds indicated in $waitForRetryTime, until the previous ones are finished
+   * or the maximun amount of tries ($maxTries) is reached. Due to the limitations of the Gearman
+   * job server, the waiting jobs will block the workers executing them until they are ended.
    */
   protected $avoidParallelExecutionJobs = array('arObjectMoveJob', 'arFileImportJob');
   protected $waitForRetryTime = 10;
+  Protected $maxRetries = 10;
```

Comment:
I thought about that, they could be overwritten in the job sub-classes to have values per job, but in the end the wait time depends more on the blocking job than on the waiting job, and that was a little more work.

This wait/retry system will only be used in sites with multiples workers available and when various sensitive jobs are running at the same time. I think we only setup a worker in our production sites at the moment, so that shouldn't be a problem.